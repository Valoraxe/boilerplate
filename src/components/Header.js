import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { startLogout } from '../actions/authentication'

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header_content">
        <Link className="header_title" to="/dashboard">
          <h1>Boilerplates</h1>
        </Link>
        <button onClick={startLogout} className="button button-link">Logout</button>
      </div>
    </div>
  </header>
)

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)
