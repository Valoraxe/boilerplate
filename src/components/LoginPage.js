import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../actions/authentication'

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout_box">
      <h1 className="box-layout_title">Boilerplate</h1>
      <p>Sub title for app</p>
      <button onClick={startLogin} className="button">Login with Google</button>
    </div>
  </div>
)

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)
