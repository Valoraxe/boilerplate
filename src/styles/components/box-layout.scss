.box-layout {
  align-items: center;
  background-image: url("/images/bg.jpg");
  background-size: cover;
  display: flex;
  justify-content: center;
  height: 100vh; //100% viewport height
  width: 100vw; //100% viewport width
}

.box-layout_box {
  background: fade-out(#fff, .15);
  border-radius: 3px;
  padding-top: $l-size;
  padding-bottom: $l-size;
  padding-left: $m-size;
  padding-right: $m-size;
  text-align: center;
  width: 25rem;
}

.box-layout_title {
  margin: 0;
  margin-bottom: $m-size;
  line-height: 1;
}
