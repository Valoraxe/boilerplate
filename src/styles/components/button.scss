.button {
  background-color: $blue;
  border: none;
  color: #ffffff;
  display: inline-block;
  font-size: 1.8rem;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button-secondary {
  background-color: #343434;
}

.button-link {
  background: none;
}
