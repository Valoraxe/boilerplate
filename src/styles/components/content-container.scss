.content-container {
  padding: 0;
  padding-left: $m-size;
  margin: 0 auto;
  max-width: 80rem;
}
