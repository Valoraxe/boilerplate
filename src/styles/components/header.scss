.header {
  background-color: $dark-blue;
}

.header_title {
  color: #ffffff;
  text-decoration: none;
  h1 {
    margin: 0;
  }
}

.header_content {
  align-items: center;
  display: flex;
  padding-top: $s-size;
  padding-bottom: $s-size;
  padding-left: 0;
  padding-right: 0;
  justify-content: space-between;
}
