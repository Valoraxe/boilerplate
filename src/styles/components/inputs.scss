.text-input {
  border: 1px solid #cacccd;
  height: 50px;
  font-size: $font-large;
  font-weight: 300;
  padding: $s-size;
}

.select {
  @extend .text-input;
}

.textarea {
  height: 10rem;
  @extend .text-input;
}
