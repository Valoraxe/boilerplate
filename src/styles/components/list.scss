.list-header {
  background: $off-white;
  border: 1px solid darken($off-white, 7%);
  color: $grey;
  display: flex;
  justify-content: space-between;
  padding-top: $s-size;
  padding-bottom: $s-size;
  padding-left: $m-size;
  padding-right: $m-size;
}

.list-item {
  border: 1px solid darken($off-white, 7%);
  border-top: none;
  color: $grey;
  display: flex;
  flex-direction: column;
  padding: $s-size;
  text-decoration: none;
  transition: background .3s ease;
  &:hover {
    background: $off-white;
  }

  @media (min-width: $desktop-breakpoint) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    padding: $m-size;
  }
}

.list-item_title {
  margin: 0;
  word-break: break-all;
}

.list-item_subtitle {
  color: $grey;
  font-size: $font-small;
}

.list-item_data {
  margin: 1.2rem 0 0 0;
  @media (min-width: $desktop-breakpoint) {
    margin: 0;
    padding-left: $s-size;
  }
}

.list-item_message {
  align-items: center;
  color: $grey;
  justify-content: center;
  padding: $m-size;
  &:hover {
    background: none;
  }
}

.list-body {
  margin-bottom: $m-size;
  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $l-size;
  }
}
