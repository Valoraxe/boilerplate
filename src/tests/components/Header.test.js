import React from 'react'
import { shallow } from 'enzyme'
import { Header } from '../../components/Header'

test("Should render correct", () => {
  const wrapper = shallow(<Header startLogout={() => { }}/>)
  expect(wrapper).toMatchSnapshot()
  //expect(wrapper.find("h1").length).toBe(1)

  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header/>)
  // expect(renderer.getRenderOutput()).toMatchSnapshot()
})

test("should call logout on click", () => {
  const startLogout = jest.fn()
  const wrapper = shallow(<Header startLogout={startLogout} />)
  wrapper.find("button").simulate("click")
  expect(startLogout).toHaveBeenCalled()
})
