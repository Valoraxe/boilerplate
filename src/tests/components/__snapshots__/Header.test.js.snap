// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correct 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header_content"
    >
      <Link
        className="header_title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boilerplates
        </h1>
      </Link>
      <button
        className="button button-link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
